### ISSUE TITLE
Always return true

### ISSUE THREAD

**Issue by fredeastside** [linslin/Yii2-Curl#22]
Hello!
When i use ```get``` method i have in an answer ```true```, but if i comment this part
```$this->setOption(CURLOPT_WRITEFUNCTION, function ($curl, $data) use (&$body) {
                $body .= $data;
                return mb_strlen($data, '8bit');
            });``` i see right answer (json).

**Comment by linslin** [comment]
Sorry, I don't understand you. Could you please send me your request Header/Body + answer Body/Header and the hole curl code you wrote?

**Comment by fredeastside** [comment]
I use this code
```
$this->curlHandle = new curl\Curl();
$this->curlHandle->setOption(CURLOPT_URL, $url);
        $this->curlHandle->setOption(CURLOPT_HTTPHEADER, [
            "Accept: application/json",
            "Content-Type: application/json",
            "Expires-at: {$expires}",
            "Signature: {$signatureCipher}",
            "Client-id: {$this->clientId}",
            "Service-secret: {$this->serviceSecret}",
        ]);
        $this->curlHandle->setOption(CURLOPT_VERBOSE, false);

        $this->curlHandle->setOption(CURLINFO_HEADER_OUT, true);
        $this->curlHandle->setOption(CURLOPT_RETURNTRANSFER, true);
```
my headers
```
GET /api/v2/countries HTTP/1.1

Host: www.XXXXXXXXXXXXXXX.com

User-Agent: Yii2-Curl-Agent

Accept: application/json

Content-Type: application/json

Expires-at: 1444844175

Signature: ZpCLeNyTW+iGwqnXXXXXXXXXXXXXXXoRthIXng==

Client-id: mYXXXXXXXXXXXXXXXihfj2A

Service-secret: DNXXXXXXXXXXXXXXXWA
```

this code ```curl_exec($curl);``` in curl\Curl return ```true```

But if I comment this lines ```$this->setOption(CURLOPT_WRITEFUNCTION, function ($curl, $data) use (&$body) {
                $body .= $data;
                return mb_strlen($data, '8bit');
            });``` in curl\Curl
i get answer (```curl_exec($curl);```) ```{"data":["AD","AE","AM","AR","AT","AU","AZ","BA","BB","BD","BE","BG","BH","BJ","BM","BN","BO","BR","BS","BY","BZ","CA","CH","CL","CM","CN","CO","CR","CY","CZ","DE","DK","DM","DO","EC","EE","EG","ES","ET","FI","FM","FR","GB","GE","GR","HK","HN","HR","HT","HU","ID","IE","IL","IN","IS","IT","JM","JO","JP","KE","KG","KH","KR","KW","KY","KZ","LB","LI","LK","LT","LU","LV","MA","MC","MD","ME","MG","MK","MN","MT","MU","MV","MX","MY","MZ","NI","NL","NO","NP","NZ","PA","PE","PG","PH","PK","PL","PR","PT","PY","QA","RO","RS","RU","SA","SC","SE","SG","SI","SK","SV","SY","SZ","TH","TJ","TM","TN","TR","TW","UA","US","UY","UZ","VE","VN","XF","XO","ZA"]}```

**Comment by linslin** [comment]
This could happend if the HTTP-Status response code isn't between  `200 && 300` or is between `400 && 510` or HTTP-Request-Method is `HEAD`. Does one of this things hit? :)

**Comment by fredeastside** [comment]
Default options
```
private $_defaultOptions = array(
        CURLOPT_USERAGENT      => 'Yii2-Curl-Agent',
        CURLOPT_TIMEOUT        => 30,
        CURLOPT_CONNECTTIMEOUT => 30,
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_HEADER         => false,
    );
```

I use get method and have status code 200.

**Comment by fredeastside** [comment]
Full code:
```
$this->curlHandle = new curl\Curl();
$this->curlHandle->setOption(CURLOPT_URL, $url);
        $this->curlHandle->setOption(CURLOPT_HTTPHEADER, [
            "Accept: application/json",
            "Content-Type: application/json",
            "Expires-at: {$expires}",
            "Signature: {$signatureCipher}",
            "Client-id: {$this->clientId}",
            "Service-secret: {$this->serviceSecret}",
        ]);
$this->curlHandle->setOption(CURLOPT_VERBOSE, false);
$this->curlHandle->setOption(CURLINFO_HEADER_OUT, true);
$this->curlHandle->setOption(CURLOPT_RETURNTRANSFER, true);
$this->curlHandle->setOption(CURLOPT_HTTPGET, true);
$response = $this->curlHandle->get($url);
```

**Comment by linslin** [comment]
Allright THX, I gonna check this!

**Comment by linslin** [comment]
As far as I could reproduce this, this bug is fixed in Version 1.0.5 @ https://github.com/linslin/Yii2-Curl/issues/19. By removing `$this->curlHandle->setOption(CURLOPT_HTTPGET, true);` in your SETUP it should work for you. Could you checkout the `DEV-Branch` and check it again?

**Comment by linslin** [comment]
I will release Version 1.0.5 today if you provide some feedback =). 

Cheers, Lin.
